### PR TITLE
Add Turkish (tr) Locale

### PR DESCRIPTION
**Pull request by lcjohnso**
Adds Turkish (tr) locale file and platform-level translated content.

Staging branch URL: https://pr-6160.pfe-preview.zooniverse.org

Project-specific test URLs: TBD

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Are the tests passing via GitHub Actions?

### CODE REVIEW

**Comment by lcjohnso** [comment]
**Significant Issue:** PFE uses `text-transform: uppercase` for multiple components, leading to the "infamous Turkish locale bug" where an uppercased version of "i" is not the correct replacement.  See [this blog entry](https://mattryall.net/blog/the-infamous-turkish-locale-bug) for a short write up.

Not sure how to proceed with this -- is there a way to hack the CSS for this particular locale?  Also acknowledging that this might require too much intervention to fix on PFE, but regardless thinking through a solution strategy would be helpful for FEM.

**Comment by eatyourgreens** [comment]
That's an interesting bug. Is it affecting the project navigation here: [Seabird Watch in Turkish](https://pr-6160.pfe-preview.zooniverse.org/projects/tr/penguintom79/seabirdwatch).

**Comment by eatyourgreens** [comment]
Also, 'Transcription' doesn't seem quite right for the Classify page there.

**Comment by lcjohnso** [comment]
Re: "transcription" for "classify" -- good to remember that our translators are often working with a particular project context in mind, and it would be good to have them review their translations on two very different project types (i.e., transcription / humanities vs. astro) for accuracy of platform-level content.

**Comment by eatyourgreens** [comment]
I see both dotted and non-dotted uppercase 'i' characters in the project navigation, but I don't know Turkish, so I can't say whether these are the correct characters or not.

Project navigation bar, showing uppercase navigation links in Turkish.

**Comment by lcjohnso** [comment]
^^ I believe the dotted uppercase letters are incorrect.

**Comment by eatyourgreens** [comment]
I found this, but it's eleven years old. [CSS text-transform not working properly for Turkish characters](https://stackoverflow.com/questions/3777443/css-text-transform-not-working-properly-for-turkish-characters).

**Comment by eatyourgreens** [comment]
iOS looks ok (mobile Safari), comparing uppercase words in the project navigation with lowercase versions of the same words further down the home page. 

Desktop Firefox looks ok too but I think it would be useful to know what the project navigation should say, so that we can fix it.

**Comment by eatyourgreens** [comment]
It might also be useful to know if this is just a problem in Chrome (and presumably Edge.)

**Comment by eatyourgreens** [comment]
@lcjohnso we already add `lang=tr` to the project page. That might be why I’m not seeing any problems in mobile Safari or Firefox. 
https://github.com/zooniverse/Panoptes-Front-End/blob/45399f3267f45fa1d67d5bcd162b4050ed11e5ac/app/pages/project/project-page.jsx#L32

If you look in the CSS, we have styles to work around bugs in Czech too. 
https://github.com/zooniverse/Panoptes-Front-End/blob/45399f3267f45fa1d67d5bcd162b4050ed11e5ac/css/typography.styl#L5-L6

**Comment by lcjohnso** [comment]
Mea culpa -- I was confused about what "correct" was supposed to look like.  I can confirm that I'm seeing consistent behavior between desktop Firefox and desktop Chrome, and in both cases I believe the navigation links are rendering correctly when upper cased: "i" -> "İ" AND "ı" -> "I".  

What confused me: seeing English words appear strangely in instances of mixed english+turkish -- example = the `<project name> Statistics` link on the project landing page.  For Seabirdwatch, it looks like this:
Screen Shot 2022-07-01 at 12 45 14 PM

Here, the issue is due to non-Turkish text mixed in ("Seabirdwatch") -- NOT incorrect capitalization transformations.

Maybe not such a big problem after all due to use of `lang=tr`!

**Comment by eatyourgreens** [comment]
Project names can be translated too eg. Every Name Counts/Jeder Name Zählt so hopefully that wouldn’t be a problem on a real Turkish project.

**Comment by eatyourgreens** [comment]
One problem we have run into in the past is the Karla font not containing the full range of Unicode characters in another language. That’s why we show Czech projects in Arial.

**Comment by lcjohnso** [comment]
Yes, Sam and I were discussing the possible need for other scripts.  But for now, I'll reach out and get the translator / relevant project team to take a look and review the content.  They are best positioned to tell us if addition work is necessary.

Also noting here: the ultimate goal is to enable Turkish translation for an FEM project (using transcription task).  Nevertheless, this PFE PR will be a first step in populating the FEM dictionaries, so worth finishing.

File: app/locales/tr.js [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+        results: 'Sonuçlar',
+        faq: 'SSS',
+        education: 'Eğitim',
+        team: 'Ekip'
+      }
+    },
+    nav: {
+      about: 'Hakkımızda',
+      adminPage: 'Admin sayfası',
+      classify: 'Transkripsiyon/Sınıflandırma',
```
Comment (goplayoutside3) [inline app/locales/tr.js:79]:
```suggestion
      classify: 'Sınıflandırma',
```

File: app/locales/tr.js [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+      title: 'Transkripsiyon.Sınıflandırma',
+      themeToggle: '%(theme)s temasına geç'
+    },
+    home: {
+      organization: 'Organizasyon',
+      researcher: 'Araştırmacıdan sözler',
+      about: '%(title)s hakkında',
+      metadata: {
+        statistics: '%(title)s İstatistikler',
+        classifications: 'Transkripsiyonlar/Sınıflandırmalar',
```
Comment (goplayoutside3) [inline app/locales/tr.js:100]:
```suggestion
        classifications: 'Sınıflandırmalar',
```

File: app/locales/tr.js [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+    },
+    loading: 'Organizasyon yükleniyor...',
+    notFound: 'Organizasyon bulunamadı',
+    notPermission: "URL'nin doğru olduğundan eminseniz, bu organizasyonu görüntüleme izniniz olmayabilir.",
+    pleaseWait: 'Lütfen bekleyiniz...',
+    stats: {
+      adjustParameters: 'Parametreleri Ayarlayın',
+      byTheNumbers: 'Sayılara göre',
+      byTheNumbersContent: {
+        classifications: 'Transkripsiyonlar/Sınıflandırmalar',
```
Comment (goplayoutside3) [inline app/locales/tr.js:156]:
```suggestion
        classifications: 'Sınıflandırmalar',
```

File: app/locales/tr.js [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+        classifications: 'Sınıflandırmalar',
+        firstProject: 'İlk Proje',
+        firstProjectLaunch: 'İlk Proje Lansman Tarihi',
+        liveProjects: 'Devam Eden Projeler',
+        pausedProjects: 'Duraklatılmış Projeler',
+        retiredProjects: 'Tamamlanmış Projeler',
+        retiredSubjects: 'Tamamlanan Temalar',
+        subjects: 'Temalar'
+      },
+      classification: 'Transkripsiyonlar/Sınıflandırmalar',
```
Comment (goplayoutside3) [inline app/locales/tr.js:165]:
```suggestion
      classification: 'Sınıflandırmalar',
```

File: app/locales/tr.js [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+      },
+      classification: 'Sınıflandırmalar',
+      comment: 'Sohbet Yorumları',
+      dateRange: 'Tarih aralığı',
+      expandWorkflowStats: 'İş Akışı İstatistiklerini Genişlet',
+      for: 'için',
+      hidden: 'İstatistikler gizlendi',
+      hourly: 'Yalnızca en son 2 haftaya ait saatlik veriler mevcuttur.',
+      organizationStatistics: '%(title)s İstatistikler',
+      perclassification: 'Transkripsiyonlar/Sınıflandırmalar ',
```
Comment (goplayoutside3) [inline app/locales/tr.js:173]:
```suggestion
      perclassification: 'Sınıflandırmalar',
```

File: app/locales/tr.js [hunk 6]
```diff
@@ -0,0 +1,471 @@
...
+      private: 'Özel',
+      submit: 'Koleksiyon Ekle'
+    }
+  },
+  emailSettings: {
+    email: 'E-posta adresi',
+    general: {
+      section: 'Zooniverse e-posta tercihleri',
+      updates: 'Genel Zooniverse e-posta güncellemelerini alın',
+      classify: 'Bir projede ilk transkripsiyon/sınıflandırma yaptığınızda e-posta güncellemesi alın',
```
Comment (goplayoutside3) [inline app/locales/tr.js:255]:
```suggestion
      classify: 'Bir projede ilk sınıflandırma yaptığınızda e-posta güncellemesi alın',
```

**Review by goplayoutside3 — APPROVED** [review]

**Comment by goplayoutside3** [comment]
> **Significant Issue:** PFE uses `text-transform: uppercase` for multiple components, leading to the "infamous Turkish locale bug" where an uppercased version of "i" is not the correct replacement. See [this blog entry](https://mattryall.net/blog/the-infamous-turkish-locale-bug) for a short write up.
> 
> Not sure how to proceed with this -- is there a way to hack the CSS for this particular locale? Also acknowledging that this might require too much intervention to fix on PFE, but regardless thinking through a solution strategy would be helpful for FEM.
> 
> EDIT: it might be possible to add a `lang` attribute that would resolve this gotcha -- see [this SO entry](https://stackoverflow.com/questions/3777443/css-text-transform-not-working-properly-for-turkish-characters).

This bug is very apparent when adding Turkish to FEM. The localization feature in FEM translates the entirety of a page, including elements in the ZooHeader not built to be translated like my username.  For example, on a Turkish FEM page my username is transformed to uppercase and looks like this: DELİLAH / NOTİFİCATİONS.

PFE doesn't have as many spots where that's an issue simply because translations weren't built to cover all components of a page, so I think it's okay for this PR to stay merged. I'm not very familiar with Turkish, but can spot one obvious incorrect word --> Any PFE project displayed in Turkish labels the classifier Tutorial tab as "TUTORİAL" because the team who submitted the PFE Turkish dictionary left that key as English "Tutorial". 

I'll look for a solution for FEM before officially enabling Turkish there, and then assess if the same solution needs to be applied to PFE.

**Pull request merged**